### PR TITLE
Documentation Update

### PR DESCRIPTION
**Pull request by tkoskela**
I've converted the documentatioin that was in `README.md`, `CONTRIBUTING.md` and `SYSTEMS.md` into a github.io website with mkdocs. I've done some reformatting and reorganisation of the sections to make the navigation more logical to me. I've also removed the other files from the root of the repository and shortened `README.md` to just contain a short header and links to the website.

There's one limitation I've come across. Mkdocs can only build the website out of files in the `docs/` directory. So I had to add links to `apps/` and `post-processing/` in `docs/` to be able to link in the markdown files in them. This approach is suggested by mkdocs e.g. [here](https://github.com/mkdocs/mkdocs/issues/1279)

### CODE REVIEW

**Comment by tkoskela** [comment]
TODO: Publish a preview page in a subdirectory on pull request

File: .github/workflows/docs_preview.yml
Comment (giordano) [inline .github/workflows/docs_preview.yml:1]: This doesn't have to be a different workflow, you can have conditionals steps: https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution

File: docs/apps.md
[the remark this answers is not in the document]
Comment (tkoskela) [reply docs/apps.md:1]: I must have added that by mistake

File: .github/workflows/docs_preview.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    name: Deploy docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@v4
+
+      - name: Add config options
+        run: |
+          echo "site_url: https://ukri-excalibur.github.io/excalibur-tests/preview" >> ./mkdocs.yml
+          echo "remote_branch: gh-preview" >> ./mkdocs.yml
```
[the remark this answers is not in the document]
Comment (tkoskela) [reply .github/workflows/docs_preview.yml]: I guess I would then have to clean up the subdirectories somehow once in a while?

[the remark this answers is not in the document]
Comment (giordano) [reply .github/workflows/docs_preview.yml]: We do that in `MPI.jl`: https://github.com/JuliaParallel/MPI.jl/blob/4ae3df31c2aa521dab8603f974363e965646c795/.github/workflows/PreviewCleanup.yml

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          fi
+          mkdocs build
+
+      - name: Deploy docs
+        run: |
+          cd gh-pages
+          mv ../site/* .
+          git add .
+          git config user.name ${{github.actor}}
+          git config user.email "${{github.actor_id}}+${{github.actor}}+@users.noreply.github.com"
```
Comment (giordano) [inline .github/workflows/docs.yml:47]: I think there's an extra `+` here:
```suggestion
          git config user.email "${{github.actor_id}}+${{github.actor}}@users.noreply.github.com"
```

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          git commit -m "Update docs"
```
Comment (giordano) [inline .github/workflows/docs.yml:48]: By adding [`github.sha`](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) to the commit message we can get a reference to the commit which trigger the build
```suggestion
          git commit -m "Update docs from ${{ github.sha }}"
```

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        run: |
+          if [[ ${{ github.event_name }} == 'pull_request' ]]; then
+            export MKDOCS_SITE_DIR="site/preview/PR${{ github.event.number }}"
+          else
+            export MKDOCS_SITE_DIR="site"
+          fi
+          mkdocs build
+
+      - name: Deploy docs
+        run: |
```
Comment (giordano) [inline .github/workflows/docs.yml:42]: This should run only if it isn't a PR from a fork (I hope I got the syntax all right!)
```suggestion
        if: ${{ github.event_name == 'push' || ! github.event.pull_request.head.repo.fork }}
        run: |
```

**Comment by tkoskela** [comment]
> Looks pretty good for the most part, but as discussed yesterday there are still a few formatting issues and hiccups with missing pages. Specifically:
> 
> * **Supported Benchmarks > Synthetic Benchmarks** - All have list formatting issues.
> * **Supported Benchmarks > Mini-apps > hpgmg** - Missing README.
> * **Supported Benchmarks > Mini-apps > sombrero** - List formatting issues.
> * **Supported Benchmarks > Applications > swift** - Missing README.
> * **Supported Systems** - It's a bit annoying that the right menu collapses whenever you click a sub-category, but otherwise not a big deal.
> * **ARCHER2 Tutorial** - Various formatting issues, particularly when displaying bash output.

I've fixed the formatting issues you found @pineapple-cat, and the most obvious ones I could find in the ARCHER2 tutorial. Thanks to @giordano suggestions you can see a preview of the deployed website [here](https://ukri-excalibur.github.io/excalibur-tests/preview/PR279/)

**Pull request merged**